### PR TITLE
Invisible reducer: Modifying state cross-reducers

### PR DESCRIPTION
**Pull request by mvbattan**
# Summary

* Introducing `Invisible Reducer` feature.
* Introducing `Wrap Service` feature: Service methods that can be dispatcheable. 
* Removing the need of using `state.merge` by default, using `configureMergeState` utility.

### What is an 'Invisible Reducer'?
An _invisible reducer_ is a reducer that holds on top of other reducers, that may modify state associated with another reducers, by overriding `combineReducers` behavior.

Although it's invisible (it does not necessarily be modified), this reducer can be modified by placing a custom reducer.

### CODE REVIEW

File: src/invisible/wrapCombineReducers/index.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    .slice(action.type.indexOf(':') + 1, action.type.indexOf('/'))
+    .replace('#', '')
+    .toLowerCase();
+  Object.keys(reducerObject).forEach(reducerName => {
+    if (reducerName.toLowerCase() === sliceName) sliceName = reducerName;
+  });
+  return sliceName;
+};
+
+const formatActionName = actionName => actionName.slice(actionName.indexOf('/') + 1);
```
Comment (ignaciosantise) [inline src/invisible/wrapCombineReducers/index.js]: should we declare `/` and `:` as constants?

File: src/invisible/wrapCombineReducers/index.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+function wrapCombineReducers(CR, invisibleReducer = commonReducer) {
```
Comment (ignaciosantise) [inline src/invisible/wrapCombineReducers/index.js]: what does `CR` mean? 🤔

Comment (mvbattan) [reply src/invisible/wrapCombineReducers/index.js]: It means `combineReducers`. It expects `redux`'s `combineReducers` to override it 😇 
If you like, I'd rename this param to make it more clear

File: src/invisible/wrapCombineReducers/index.js (hunk 1 above)
Comment (mvbattan) [reply src/invisible/wrapCombineReducers/index.js]: Makes sense !

**Comment by WFolini** [comment]
@mvbattan I put a calendar event to look at it tonight 👼

**Pull request merged**